packages/components/src/components/input/input.ts

```diff
@@ -388,61 +403,129 @@ export default class SdInput extends SolidElement implements SolidFormControl {
   }
 
   render() {
-    const hasLabelSlot = this.hasSlotController.test('label');
-    const hasHelpTextSlot = this.hasSlotController.test('help-text');
-    const hasLabel = this.label ? true : !!hasLabelSlot;
-    const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
-    const hasClearIcon =
-      this.clearable && !this.disabled && !this.readonly && (typeof this.value === 'number' || this.value.length > 0);
-
+    // Slots
+    const slots = {
+      label: this.hasSlotController.test('label'),
+      helpText: this.hasSlotController.test('help-text'),
+      description: this.hasSlotController.test('description'),
+      left: this.hasSlotController.test('left'),
+      right: this.hasSlotController.test('right')
+    };
+
+    // States
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
+    const hasClearIcon = this.clearable && !this.readonly && (typeof this.value === 'number' || this.value.length > 0);
+    const hasValidationAttr = this.required || !!this.pattern || !!this.minlength || !!this.maxlength;
+    const isInvalid = hasValidationAttr && !this.checkValidity();
+    const isValid = hasValidationAttr && this.checkValidity();
+    // Hierarchy of input states:
+    const inputState = this.disabled
+      ? 'disabled'
+      : this.readonly
+      ? 'readonly'
+      : this.hasFocus && isInvalid
+      ? 'activeInvalid'
+      : this.hasFocus && isValid
+      ? 'activeValid'
+      : this.hasFocus
+      ? 'active'
+      : isInvalid
+      ? 'invalid'
+      : isValid
+      ? 'valid'
+      : 'default';
+
+    // Conditional Styles
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[inputState];
+
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[inputState];
+
+    const iconMarginLeft = this.size === 'sm' ? 'ml-1' : 'ml-2';
```

Comment:
Please write iconMarginLeft like you did in iconSize:
`const iconMarginLeft = {
      sm: 'ml-1',
      md: 'ml-2',
      lg: 'ml-2'
    }[this.size];`